src/core/dbus-slice.c

```diff
@@ -1,15 +1,64 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+#include "bus-get-properties.h"
 #include "dbus-cgroup.h"
 #include "dbus-slice.h"
+#include "dbus-util.h"
 #include "slice.h"
 #include "unit.h"
 
+static int property_get_currently_active(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        Slice *s = ASSERT_PTR(userdata);
+
+        assert(bus);
+        assert(reply);
+
+        return sd_bus_message_append(
+                        reply,
+                        "u",
+                        (uint32_t) slice_get_currently_active(s, /* ignore= */ NULL, /* with_pending= */ false));
+}
+
 const sd_bus_vtable bus_slice_vtable[] = {
         SD_BUS_VTABLE_START(0),
+        SD_BUS_PROPERTY("ConcurrencyHardMax", "u", bus_property_get_unsigned, offsetof(Slice, concurrency_hard_max), 0),
+        SD_BUS_PROPERTY("ConcurrencySoftMax", "u", bus_property_get_unsigned, offsetof(Slice, concurrency_soft_max), 0),
+        SD_BUS_PROPERTY("NCurrentlyActive", "u", property_get_currently_active, 0, 0),
         SD_BUS_VTABLE_END
 };
 
+static int bus_slice_set_transient_property(
+                Slice *s,
+                const char *name,
+                sd_bus_message *message,
+                UnitWriteFlags flags,
+                sd_bus_error *error) {
+
+        Unit *u = UNIT(s);
+
+        assert(s);
+        assert(name);
+        assert(message);
+
+        flags |= UNIT_PRIVATE;
+
+        if (streq(name, "ConcurrencyHardMax"))
+                return bus_set_transient_unsigned(u, name, &s->concurrency_hard_max, message, flags, error);
+
+        if (streq(name, "ConcurrencySoftMax"))
+                return bus_set_transient_unsigned(u, name, &s->concurrency_soft_max, message, flags, error);
+
```

Comment:
It'd make sense to dispatch jobs if soft limit is increased here.
But then it'd also make sense to cull jobs/units if hard/soft limit is decreased.

Reply:
the current PR doesn't allow this to be changed during runtime (i.e. systemctl set-property is not supported for it). we should look into changing that, but let's do that later.

(in particular to allow a scheme where you initially set ConcurrencySoftMax to 0, and then at some later point issue set-property to let the queue to be processed)

Reply:
I commented to DBus line but it'd actually pertain to parse config + daemon-reload too. So the implementation should tackle it somehow.
(Or take the simpler path and accept only changes before slice is active, similar to how `unit_set_slice()` rejects changes.)

Reply:
I think eventually we want the props to be fully dynamic. i wanted to keep this short, hence didn't implement this for now, but i think the goal must be clear.

and i think a pattern where you define a slice with a limit of zero initially, and then eventually "unblock" by bumping it is a completely ok scenario. (the opposite is ok too: you allow a non-zero amount of concurrency, and then you set the limit dynamically to zero, to temporarily block further jobs being added).

the unit file stores the original value of course, by raising or lowering the limits via dbus we can then do the unblock/block stuff